### PR TITLE
S3 intermediates manager

### PR DESCRIPTION
**Pull request by mgasner**

### CODE REVIEW

File: python_modules/dagster/dagster/core/execution.py [hunk 1]
```diff
@@ -379,6 +379,9 @@ def construct_run_storage(run_config, environment_config):
         return FileSystemRunStorage()
     elif environment_config.storage.storage_mode == 'in_memory':
         return InMemoryRunStorage()
+    elif environment_config.storage.storage_mode == 's3':
+        # TODO: Revisit whether we want to use S3 run storage
```
Comment (schrockn) [inline python_modules/dagster/dagster/core/execution.py]: ?????

Comment (schrockn) [reply python_modules/dagster/dagster/core/execution.py]: ok nvm i understand

**Comment by mgasner** [comment]
very clean, thanks for the abstraction layers :)

**Comment by mgasner** [comment]
@schrockn confused why we don't need to do https://github.com/dagster-io/dagster/pull/970/files#diff-934977d8a5d332fb279f3e603ed34d41R332 in the filesystem-based tests. afaict, we should be throwing at https://github.com/dagster-io/dagster/blob/260dcdad73e780169a52091f791c55e8a597d5e2/python_modules/dagster/dagster_tests/core_tests/test_external_execution_plan.py#L269 -- or is it that we are missing a check in https://github.com/dagster-io/dagster/blob/260dcdad73e780169a52091f791c55e8a597d5e2/python_modules/dagster/dagster/core/object_store.py#L76

**Pull request merged**